From aa6a1a562533ddda69e67279241266ca05c937fd Mon Sep 17 00:00:00 2001
From: Martin Fick <mfick@codeaurora.org>
Date: Mon, 19 Mar 2012 13:00:18 -0600
Subject: [PATCH] Unify issue processing and make more robust

The issue processing in gitlog2asciidoc.py missed
certain capitilazation combinations and was split
across 2 ifs.  Combine it into one if and make it case
insensitive using an ugly python(pre 2.7) backwards
compatible method.

Change-Id: I49fb4d66bd5b3777cc9641a947730a19222895f6
---
 tools/gitlog2asciidoc.py |    8 ++------
 1 files changed, 2 insertions(+), 6 deletions(-)

diff --git a/tools/gitlog2asciidoc.py b/tools/gitlog2asciidoc.py
index 6f063b2..3080289 100755
--- a/tools/gitlog2asciidoc.py
+++ b/tools/gitlog2asciidoc.py
@@ -62,12 +62,8 @@ for line in stdout_value.splitlines(True):
         continue
 
     # Move issue number to subject line
-    elif re.match('Bug: ', line) is not None:
-        line = line.replace('Bug: ', '').replace('\n',' ')
-        subject = subject[:2] + line + subject[2:]
-    # Move issue number to subject line
-    elif re.match('Issue: ', line) is not None:
-        line = line.replace('Issue: ', 'issue ').replace('\n',' ')
+    elif re.match('([bB][uU][gG]|[iI][sS][sS][uU][eE]):( [iI][sS][sS][uU][eE])? ', line) is not None:
+        line = re.sub('([bB][uU][gG]|[iI][sS][sS][uU][eE]):( [iI][sS][sS][uU][eE])? ', 'issue ', line, re.I).replace('\n',' ')
         subject = subject[:2] + line + subject[2:]
 
     # Remove commit footers
-- 
1.7.8

